### PR TITLE
[DON'T MERGE][BUILD] Switch default protobuf-java version to 3.x

### PR DESCRIPTION
**Pull request by LuciferYang**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -827,8 +827,7 @@
       <dependency>
         <groupId>com.google.protobuf</groupId>
         <artifactId>protobuf-java</artifactId>
-        <version>${protobuf.hadoopDependency.version}</version>
-        <scope>${hadoop.deps.scope}</scope>
+        <version>${protobuf.version}</version>
```
Comment (LuciferYang) [inline pom.xml:830]: I have done maven test with `-Phadoop-2` and `-Phadoop-3`, and all tests passed.

If feasible, we don't need to explicitly add `protobuf-java` dependency and declare `protobuf.version` in each module that requires `protobuf-java` 3.x(such as core, sql, connect and protobuf), but this also changes `protobuf-java` version in `dev/deps/spark-deps-hadoop-x-hive-2.3` from 2.5.0 to 3.x

Comment (LuciferYang) [reply pom.xml:823]: Of course, it may be safer to remove `-Phadoop-2` completely before doing this work

**Comment by bjornjorgensen** [comment]
Are there anything we can do with this one? 
https://github.com/apache/spark/blob/267b50f0392bcead3a8bd39c6b7ad9ce7a468d44/connector/kinesis-asl-assembly/pom.xml#L65

**Comment by bjornjorgensen** [comment]
oh.. https://github.com/apache/spark/pull/33133 the Guava problem..

**Comment by LuciferYang** [comment]
@bjornjorgensen The pr is still being tested. I feel strange that the yarn module can pass the test with `-Phadoop-2`.

**Pull request closed without merging**